compiler/test/dshell/dll.d

```diff
@@ -6,8 +6,11 @@ int main()
         return DISABLED;
 
     version (DigitalMars)
-        version (OSX) // Shared libraries are not yet supported on OSX
+    {
+// Disable DM Dlls for now, need to redesign it
```

Comment:
So this PR doesn't add a working test, it only removes a test. I'm fine with that, under the presumption that DMD dlls are so broken currently that nobody uses them.